### PR TITLE
Register application "See more", "Home", "Create" and "Workspace Settings"

### PR DESCRIPTION
**Pull request by yuye-aws**
### Description

In this PR, I registered four workspace related applications: list, create, update and overview. Users can go to overview page by "Home" link in left menu, and update the workspace setting in "Workspace Settings" link in left menu. Please note that there are two problems:
1. "Home" and "Workspace Settings" are still available when outside workspace. 
2. All nav links are visible when inside workspace.
I will address these issues in another PR, which targets to filter nav links according to categories
.
## Screenshot

Outside workspace:
image

### Check List

- [ ] All tests pass
  - [ ] `yarn test:jest`
  - [ ] `yarn test:jest_integration`
  - [ ] `yarn test:ftr`
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
- [ ] Update [CHANGELOG.md](./../CHANGELOG.md)
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: src/plugins/workspace/public/application.tsx [hunk 1]
```diff
@@ -6,19 +6,74 @@
...
+import { WorkspaceCreatorApp } from './components/workspace_creator_app';
+import { WorkspaceUpdaterApp } from './components/workspace_updater_app';
+import { WorkspaceOverviewApp } from './components/workspace_overview_app';
+
+export const renderListApp = (
+  { element, history, appBasePath }: AppMountParameters,
+  services: CoreStart
+) => {
+  ReactDOM.render(
+    <Router history={history}>
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/application.tsx:21]: As we are using applications as our router, is the Router component essential for the workspace app to run?

File: src/plugins/workspace/public/plugin.ts [hunk 2]
```diff
@@ -109,19 +167,21 @@ export class WorkspacesPlugin implements Plugin<{}, {}, WorkspacesPluginSetupDep
   private workspaceToChromeNavLink(
     workspace: WorkspaceAttribute,
     workspacesStart: WorkspacesStart,
-    application: ApplicationStart
+    application: ApplicationStart,
+    index: number
   ): ChromeNavLink {
     const id = WORKSPACE_APP_ID + '/' + workspace.id;
     const url = workspacesStart?.formatUrlWithWorkspaceId(
       application.getUrlForApp(WORKSPACE_APP_ID, {
```
Comment (SuZhou-Joe) [inline src/plugins/workspace/public/plugin.ts:175]:
```suggestion
      application.getUrlForApp(WORKSPACE_APP_ID + PATHS.overview, {
       absolute: true,
     }),
```

We should use the new applicationId in this case I think?

**Review by SuZhou-Joe — APPROVED** [review]

File: src/plugins/workspace/public/components/workspace_updater_app.tsx [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+   * map the current pathname to breadcrumbs
+   */
+  useEffect(() => {
+    const breadcrumbs = createBreadcrumbsFromPath(location.pathname, ROUTES, appBasePath);
+    chrome?.setBreadcrumbs(breadcrumbs);
+  }, [appBasePath, location.pathname, chrome]);
+
+  return (
+    <I18nProvider>
+      <WorkspaceUpdater />
```
Comment (ruanyl) [inline src/plugins/workspace/public/components/workspace_updater_app.tsx:30]: If updating breadcrumbs is not needed, I think we can just use `WorkspaceUpdater` in `renderUpdateApp`, no need to have `WorkspaceUpdaterApp`

File: src/plugins/workspace/public/plugin.ts [hunk 4]
```diff
@@ -84,22 +90,70 @@ export class WorkspacesPlugin implements Plugin<{}, {}, WorkspacesPluginSetupDep
      */
     savedObjectsManagement?.columns.register(getWorkspaceColumn(core));
 
+    type WorkspaceAppType = (params: AppMountParameters, services: CoreStart) => () => void;
+    const workspaceMount = async (params: AppMountParameters, renderApp: WorkspaceAppType) => {
```
Comment (ruanyl) [inline src/plugins/workspace/public/plugin.ts:94]: Nit: I think we can call it `mountWorkspaceApp`

**Review by ruanyl — APPROVED** [review]

**Pull request merged**